packages/cozy-sharing/src/helpers/successMessage.js

```diff
@@ -1,16 +1,39 @@
 import { models } from 'cozy-client'
 
-import { Contact } from '../models'
+import { Contact, Group } from '../models'
+
 export const countNewRecipients = (currentRecipients, newRecipients) => {
-  return newRecipients.filter(contact => {
-    const email = models.contact.getPrimaryEmail(contact)
-    const cozyUrl = models.contact.getPrimaryCozy(contact)
-    return !currentRecipients.find(
-      r =>
-        (email && r.email && r.email === email) ||
-        (cozyUrl && r.instance && r.instance === cozyUrl)
-    )
-  }).length
+  return newRecipients.reduce(
```

Comment:
nit: If we assume that a sharing always have a few recipients, maybe we can sacrifice a bit of performance and improve readability by using two `filter` loops instead of a single `reduce`?

Reply:
Yes good point I split the filter step into its own before the reduce 👍